src/fmeval/eval_algorithms/answer_relevance.py

```diff
@@ -0,0 +1,289 @@
+        self.model_output_key = model_output_key
+        self.embeddings_model = embeddings_model
+
+    @validate_call
+    def __call__(self, record: Dict[str, Any]) -> Dict[str, Any]:
+        """Augment the input record with the computed answer relevance score.
+
+        :param record: The input record.
+        :returns: The input record, with the answer relevance score added in.
+        """
+        gen_questions = record["gen_questions"].split("\n")
+        question = record[self.model_output_key]
+        record[self.output_key] = self._get_score(self.embeddings_model, question, gen_questions)
+        return record
+
+    @staticmethod
+    def _get_score(embeddings_model: ModelRunner, question: str, generated_questions: List[str]) -> float:
+        """Given generated questions and the original question, embedding model, computer Answer Relevance score.
+
+        :param embeddings_model: An instance of ModelRunner representing the embedding model to be used.
+        :param question: The given prompt (model input).
+        :param generated_questions: An artificial question generated by judge model based on the answer.
+        :returns: 0 to 1. See the docstring for `Answer Relevance` for more details
+            on what these numerical values represent.
+        """
+        question_vector = np.asarray(embeddings_model.predict(question))
+        gen_question_vectors = []
+        for generated_question in generated_questions:
+            gen_question_vector = np.asarray(embeddings_model.predict(generated_question))
+            gen_question_vectors.append(gen_question_vector)
+        norm = np.linalg.norm(gen_question_vectors) * np.linalg.norm(question_vector)
+        return np.dot(gen_question_vector, question_vector.T) / norm
+
+
+class GenerateQuestions(Transform):
+    """This transform invokes a judge model to generate questions."""
+
+    def __init__(self, input_key: str, output_key: str, judge_model: ModelRunner):
+        """GenerateQuestions Initializer.
+        :param input_key: The key corresponding to prompt to get statements.
+        :param output_key: The key corresponding to the statements, which gets added to the record.
+        :param judge_model: An instance of ModelRunner representing the judge model to be used.
+        """
+        super().__init__(input_key, output_key, judge_model)
+        self.register_input_output_keys(
+            input_keys=[input_key],
+            output_keys=[output_key],
+        )
+        self.input_key = input_key
+        self.output_key = output_key
+        self.judge_model = judge_model
+
+    @validate_call
+    def __call__(self, record: Dict[str, Any]) -> Dict[str, Any]:
+        """Invokes the judge model to generate statements and augments the input record with the statements.
+        :param record: The input record.
+        :returns: The input record with the statements added in.
+        """
+        gen_raw_questions = GetModelOutputs(
```

Comment:
One argument against creating other `Transform` objects inside another Transform's `__call__` method is that you need to be very careful about ensuring that the input keys for the "child transform" (the one nested inside `__call__`) are reflected in the parent transform's input keys in `self.register_input_output_keys`.

This makes the code more confusing imo and can lead to bugs pretty easily. I would suggest taking out the `GetModelOutputs` and placing it in the pipeline for the eval algo instead. You can think of this as being similar to dependency injection in a sense.